src/dialogs/config-flow/dialog-data-entry-flow.ts

```diff
@@ -295,13 +305,10 @@ class DataEntryFlowDialog extends LitElement {
     }
   }
 
-  private _close(): void {
-    if (this._step) {
-      this._flowDone();
-    } else if (this._step === null) {
-      // Flow aborted during picking flow
-      this._step = undefined;
-      this._params = undefined;
+  private _close(ev: CustomEvent) {
+    // If closed is called because of disconnect of ha-dialog, we don't have to handle
+    if ((ev.target as HaDialog).isConnected) {
+      history.back();
```

Comment:
`history.back` seems weird here. The dialog should not be aware on how these things work. Maybe we can add an event to dialogs to indicate when they are closing?